dts/bindings/dma/st,stm32-dma.yaml

```diff
@@ -18,6 +18,14 @@ properties:
       type: boolean
       description: If the controller supports memory to memory transfer
 
+    st,periph2mem:
+      type: boolean
+      description: If the controller supports pheripheral to memory transfer
+
+    st,mem2periph:
+      type: boolean
+      description: If the controller supports memory to pheripheral transfer
+
```

Comment:
This addition looks not specific to DMAMUX, if true, put this in a dedicated commit (prior to dmamux ones if necessary for dmamux).
Also, existing "st,mem2mem" was currently used to assert support of mem2mem operation. 
Support of mem2periph and periph2mem were implicit, and assumed this way in the driver.
Adding  "st,periph2mem" and "st,mem2periph" is making support of these capability explicit, which, I guess, has impact on the driver. This impact has to be put in a dedicated commit, prior to pure DMAMUX related changes.

Reply:
We will keep st,mem2mem if the dma instance ONLY supports memory to memory transfer (ex. DMA2 of the stm32F4xx). For any other instance (all transfer directions are possible), the st,mem2mem is not set.

Reply:
for L4 series, the dma should not include the mem2mem (dt/arm/st/l4/stm32l4r5.dtsi)

Reply:
changed:  except for DMA V1, where mem2mem is a limitation